dev/fattest.simplicity/src/componenttest/topology/utils/ExternalTestServiceDecrypter.java

```diff
@@ -0,0 +1,278 @@
+        ExternalTestService instance = null;
+        while (instance == null) {
+            try {
+                instance = ExternalTestService.getService("liberty-properties-decrypter");
+            } catch (Exception e) {
+                if (DECRYPTER_SERVICE_RETRY.retryable("RE_INITALIZER", e.getMessage())) { // Wait and continue
+                    continue;
+                } else {
+                    throw new RuntimeException("Could not get a liberty-properties-decrypter service", e);
+                }
+            }
+        }
+        return instance;
+    };
+
+    /**
+     * Uses the decrypter external test service to decrypt the provided property
+     *
+     * @param  property              the property to be decrypted
+     * @return                       returns the original property if the property was not encrypted,
+     *                               otherwise returns the decrypted property.
+     *
+     * @throws IllegalStateException if the access token was not set, or if the decypter service rejected the token
+     * @throws NullPointerException  if the decrypter service was was unreachable after a series of retries
+     */
+    public static String decrypt(String property) {
+        final String m = "decrypt";
+
+        // Do not decrypt property if it is not encrypted
+        if (!property.startsWith(ENCRYPTED_PREFIX)) {
+            return property;
+        }
+
+        // Ensure access token is available
+        final String accessToken = findAccessToken();
+
+        // Get a decrypter service, new or existing
+        ExternalTestService decrypter = DECRYPTER_SERVICE.computeIfAbsent(SERVICE_KEY, INITALIZER);
+
+        // Set retry policy for decrypting a property, to account for occasional network issues.
+        final RetryPolicy retry = new CounterRetryPolicy(5);
+
+        String decrypted = null;
+        while (decrypted == null) {
+            // Create request to decrypter service
+            final String URL = "https://" + decrypter.getAddress() + ":" + decrypter.getPort() + "/decrypt?value=" + property + "&access_token=" + accessToken;
+            final HttpsRequest request = new HttpsRequest(URL)
+                            .allowInsecure()
+                            .timeout(10000)
+                            .expectCode(HttpsURLConnection.HTTP_OK)
+                            .expectCode(HttpsURLConnection.HTTP_UNAUTHORIZED)
+                            .expectCode(HttpsURLConnection.HTTP_FORBIDDEN)
+                            .silent();
+
+            try {
+                decrypted = request.run(String.class);
+            } catch (Exception e) {
+
+                // The decrypter threw an unexpected exception, mark it unhealthy
+                ExternalTestServiceReporter.reportUnhealthy(decrypter, e);
+
+                //Should we retry
+                if (retry.retryable(m, e.toString())) {
```

Comment:
Perhaps it would be more efficient to have a `willRetry()` method for the `DECRYPTER_SERVICE_RETRY` RetryPolicy so we don't attempt to get a new service if we know we will not retry.

```suggestion
                if (retry.retryable(m, e.toString()) && DECRYPTER_SERVICE_RETRY.willRetry()) {
```